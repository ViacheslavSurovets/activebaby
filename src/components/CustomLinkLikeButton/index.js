export { default as CustomLinkLikeButton } from './CustomLinkLikeButton';
