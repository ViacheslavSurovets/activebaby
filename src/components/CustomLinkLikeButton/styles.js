import { Link } from 'react-router-dom';
import styled, { css } from 'styled-components';
import { commonFontSize, flexCommon } from '@commonStyles';

const stylesLinkLikeButtonDefault = css`
  ${flexCommon};
  ${ commonFontSize };
  color: ${ ( { theme } ) => theme.customButton.customButtonColor };
  border: 1px solid ${ ( { theme } ) => theme.customButton.customButtonBorderColor };
  background-color: ${ ( { theme } ) => theme.white };
  outline: none;
  text-decoration: none;
  width: 10rem;
  height: 5rem;
  &:hover {
    background: ${ ( { theme } ) => theme.black };
    color: ${ ( { theme } ) => theme.white };
  }
`;


export const LinkButton = styled ( Link )`
    ${ stylesLinkLikeButtonDefault };
`;
