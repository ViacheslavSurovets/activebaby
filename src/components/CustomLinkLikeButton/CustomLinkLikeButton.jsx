import React from 'react';
import { LinkButton } from './styles';
import PropTypes from 'prop-types';

const CustomLinkLikeButton = ({ children }, url,  { ...rest } ) => (
  <LinkButton to={ url } { ...rest }>
    { children }
  </LinkButton>
);

export default CustomLinkLikeButton;

CustomLinkLikeButton.propTypes = {
  children: PropTypes.string
};
