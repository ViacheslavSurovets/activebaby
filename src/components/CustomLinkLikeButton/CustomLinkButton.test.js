import React from 'react';
import { shallow } from 'enzyme';
import CustomLinkLikeButton from './CustomLinkLikeButton';


describe ( 'CustomLinkLikeButton', () => {
  it ( 'CustomLinkLikeButton Snapshot', () => {
    expect ( shallow ( <CustomLinkLikeButton /> ) ).toMatchSnapshot ();
  } );
} );
