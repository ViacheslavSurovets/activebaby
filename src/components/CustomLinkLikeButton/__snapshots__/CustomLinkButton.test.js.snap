// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomLinkLikeButton CustomLinkLikeButton Snapshot 1`] = `ShallowWrapper {}`;
